app-shell/src/app/shell-parser/template-parser/parse5/tokenizer-patch.ts

```diff
@@ -0,0 +1,42 @@
+var Tokenizer = require('../../../../vendor/parse5/lib/tokenizer');
+
+// Monkey patching the lexer in order to establish
+// case sensitive parsing of the input templates.
+// This way we'll be able to use case sensitive attribute
+// and element selectors for stripping the content that
+// is not required for the App Shell.
+//
+// Since we're patching module's internals we cannot
+// use parse5 as dependency of the App Shell since we
+// won't have access to the tokenizer in order to patch
+// it runtime. Because of that we distribute the entire
+// Runtime Parser as a single bundle which includes parse5.
+Tokenizer.prototype.getNextToken = function () {
+  function replaceLastWithUppercase(token: any, prop: string, cp: number) {
+    if (token) {
+      let char = String.fromCharCode(cp);
+      let val = token[prop];
+      let last = val[val.length - 1];
+      if (last !== char) {
+        token[prop] = val.substring(0, val.length - 1) + last.toUpperCase();
```

Comment:
Tests get errors on this line if last is undefined (`TypeError: Cannot read property 'toUpperCase' of undefined`). If I change to `if (last && last !== char)` then tests pass. Safe change to make?